src/components/ArticlesData.jsx

```diff
@@ -1,33 +1,47 @@
 import { useEffect, useState } from "react"
-import ArticlesList from "./ArticlesList"
 import ApiRequest from "./ApiRequest"
+import ArticlesList from "./ArticlesList"
 
-export default function ArticlesDataHolder() {
+export default function ArticlesData({ article_id } = null) {
     const [articles, setArticles] = useState([])
     const [isLoading, setIsLoading] = useState(false)
-    const [isError, setIsError] = useState(false)
+    const [errMsg, setErrMsg] = useState(null)
 
     useEffect(() => {
+        let url = `/articles`
+
+        if (article_id) {

```

Comment:
I would have a separate call for articles and article by article_id - itll be easier for when there are more options eg topic, order by